kmp-sharing/src/iosMain/kotlin/com/swmansion/kmpsharing/Sharing.kt

```diff
@@ -0,0 +1,50 @@
+package com.swmansion.kmpsharing
+
+import kotlinx.cinterop.*
+import platform.Foundation.*
+import platform.UIKit.*
+
+public actual class Sharing {
+    public actual companion object {
+        @OptIn(ExperimentalForeignApi::class)
+        public actual fun share(url: String, options: SharingOptions?) {
+            try {
+                val nsUrl =
+                    NSURL.URLWithString(url)
+                        ?: throw SharingInvalidArgumentException("Invalid URL: $url")
+
+                val activityItems = listOf(nsUrl)
+                val activityViewController =
+                    UIActivityViewController(
+                        activityItems = activityItems,
+                        applicationActivities = null,
+                    )
+
+                options?.iosAnchor?.let { anchor ->
+                    activityViewController.popoverPresentationController?.let { popover ->
+                        popover.sourceView = UIApplication.sharedApplication.keyWindow
+                        popover.sourceRect =
+                            platform.CoreGraphics.CGRectMake(
+                                anchor.x.toDouble(),
+                                anchor.y.toDouble(),
+                                anchor.width.toDouble(),
+                                anchor.height.toDouble(),
+                            )
+                    }
+                }
+
+                val rootViewController =
+                    UIApplication.sharedApplication.keyWindow?.rootViewController
+                        ?: throw SharingFailedException("Could not find root view controller")
```

Comment:
isn't this like a super pathological case that should just be a plain `RuntimeException` ?